tests/drivers/hwinfo/reset_cause/src/main.c

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/sys/reboot.h>
+#include <errno.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(resetreason, LOG_LEVEL_INF);
+
+static const struct device *const my_wdt_device = DEVICE_DT_GET(DT_ALIAS(watchdog0));
+static struct wdt_timeout_cfg m_cfg_wdt0;
+static int my_wdt_channel;
+
+#if DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
```

Comment:
Does this test need using dtcm section?

Reply:
I just copied this code from watchdog driver tests:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c#L151

DTCM seems to be available on Cortex-M cores:
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/arm/arm,dtcm.yaml

Shall I remove this #if or can it stay? Currently, platform_allow list doesn't contain any target with Cortex-M core.